http/src/client/mod.rs

```diff
@@ -512,35 +512,41 @@ impl Client {
     ///
     /// # Examples:
     ///
-    /// Create permission overrides for a role to view the channel, but not send messages:
+    /// Create permission overrides for a role to view the channel, but not send
+    /// messages:
     ///
     /// ```no_run
+    /// # #[tokio::main] async fn main() -> Result<(), Box<dyn std::error::Error>> {
     /// # use twilight_http::Client;
-    /// use twilight_model::guild::Permissions;
-    /// use twilight_model::id::Id;
-    /// #
-    /// # #[tokio::main]
-    /// # async fn main() -> Result<(), Box<dyn std::error::Error>> {
     /// # let client = Client::new("my token".to_owned());
+    /// #
+    /// use twilight_model::{
+    ///     guild::Permissions,
+    ///     http::permission_overwrite::{
+    ///         PermissionOverwrite, PermissionOverwriteType,
+    ///     },
+    ///     id::Id,
+    /// };
     ///
     /// let channel_id = Id::new(123);
-    /// let allow = Permissions::VIEW_CHANNEL;
-    /// let deny = Permissions::SEND_MESSAGES;
-    /// let role_id = Id::new(432);
+    /// let permission_overwrite = PermissionOverwrite {
+    ///     allow: Some(Permissions::VIEW_CHANNEL),
+    ///     deny: Some(Permissions::SEND_MESSAGES),
+    ///     id: Id::new(432),
+    ///     kind: PermissionOverwriteType::Role,
+    /// };
```

Comment:
This is what I meant when I said I would like to have a example use cast as that is likely what you will want to use often
```suggestion
    /// let role_id: Id<RoleMarker> = Id::new(432);
    /// let permission_overwrite = PermissionOverwrite {
    ///     allow: Some(Permissions::VIEW_CHANNEL),
    ///     deny: Some(Permissions::SEND_MESSAGES),
    ///     id: role_id.cast(),
    ///     kind: PermissionOverwriteType::Role,
    /// };
```